micro-rdk/src/common/ota.rs

```diff
@@ -304,11 +302,18 @@ impl<S: OtaMetadataStorage> OtaService<S> {
             uri = hyper::Uri::from_parts(parts).map_err(|e| OtaError::Other(e.to_string()))?;
         };
 
+        let mut num_tries = 0;
         let (mut sender, conn) = loop {
+            if num_tries == NUM_RETRY_CONN {
```

Comment:
Is this a necessary part of this change? It seems independent of moving the driver of OTA checks to be the ConfigMonitor.